pac4j-oauth/src/main/java/org/pac4j/oauth/profile/generic/GenericOAuth20ProfileDefinition.java

```diff
@@ -0,0 +1,117 @@
+package org.pac4j.oauth.profile.generic;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.Verb;
+import java.util.HashMap;
+import java.util.Map;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.core.profile.converter.AttributeConverter;
+import org.pac4j.core.profile.converter.StringConverter;
+import org.pac4j.oauth.config.OAuth20Configuration;
+import org.pac4j.oauth.profile.JsonHelper;
+import org.pac4j.oauth.profile.definition.OAuth20ProfileDefinition;
+
+/**
+ * <p>This class is the user profile for generic OAuth2 with appropriate getters.</p>
+ * <p>The map of <code>profileAttributes</code> is intended to replate the primary/secondary attributes where
+ * the key is the name of the attribute and the value is the path to obtain that attribute from the
+ * json resopnse, starting from <code>firstNodePath</code></p>
+ * <p>It is returned by the {@link org.pac4j.oauth.client.GenericOAuth2Client}.</p>
+ *
+ * @author Julio Arrebola
+ */
+public class GenericOAuth20ProfileDefinition extends OAuth20ProfileDefinition<GenericOAuth20Profile> {
+
+    
+    private final Map<String,String> profileAttributes = new HashMap<>();
+    
+    String profileUrl = null;
+    String profileMethod = null;
+    String firstNodePath = null;
+    
+    public void setProfileMethod(String value) {
+        this.profileMethod = value;
+    }
+
+    @Override
+    public Verb getProfileVerb() {
```

Comment:
I guess you mean the name of the function should be "getProfileMethod" instead of "getProfileVerb". I think so too. And I just realized I forgot to make this method configurable.